// hibp/dist/cjs/api/haveibeenpwned/responses.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

/**
 * Known potential responses from the remote API.
 *
 * Unfortunately, the API does not send a decent human-readable message back
 * with each response, but they are documented on the website:
 * https://haveibeenpwned.com/api/v3#ResponseCodes
 *
 * These objects simply provide a mapping between the HTTP response status code
 * and the corresponding human-readable message so we can throw a more
 * descriptive error for the consumer. (They are also leveraged in our tests.)
 */
var emptyHeaders = /*#__PURE__*/new Map();
/** @internal */

var OK = {
  headers: emptyHeaders,
  status: 200
};
/** @internal */

var BAD_REQUEST = {
  headers: emptyHeaders,
  status: 400,
  statusText: 'Bad request — the account does not comply with an acceptable format.'
};
/**
 * This response has unique behavior. For some reason, the API includes an
 * object in the response body for this one, containing a human-readable
 * message. Manually populating the message here purely for use in tests.
 *
 * @internal
 */

var UNAUTHORIZED = {
  headers: emptyHeaders,
  status: 401,
  body: {
    statusCode: 401,
    message: 'Access denied due to missing hibp-api-key.'
  }
};
/** @internal */

var FORBIDDEN = {
  headers: emptyHeaders,
  status: 403,
  statusText: 'Forbidden - access denied.'
};
/** @internal */

var BLOCKED = {
  headers: /*#__PURE__*/new Map([['cf-ray', 'someRayId']]),
  status: 403
};
/** @internal */

var NOT_FOUND = {
  headers: emptyHeaders,
  status: 404
};
/**
 * This response has unique behavior. For some reason, the API includes an
 * object in the response body for this one, containing a human-readable
 * message. Manually populating the message here purely for use in tests.
 *
 * @internal
 */

var TOO_MANY_REQUESTS = {
  headers: emptyHeaders,
  status: 429,
  body: {
    statusCode: 429,
    message: 'Rate limit is exceeded. Try again in 2 seconds.'
  }
};

exports.BAD_REQUEST = BAD_REQUEST;
exports.BLOCKED = BLOCKED;
exports.FORBIDDEN = FORBIDDEN;
exports.NOT_FOUND = NOT_FOUND;
exports.OK = OK;
exports.TOO_MANY_REQUESTS = TOO_MANY_REQUESTS;
exports.UNAUTHORIZED = UNAUTHORIZED;
